packages/cozy-procedures/src/components/documents/EmptyDocumentHolder.jsx

```diff
@@ -1,67 +1,142 @@
 import DocumentsDataFormContainer from '../../containers/DocumentsDataForm'
-const EmptyDocumentHolder = ({
-  t,
-  documentId,
-  client,
-  linkDocumentSuccess
-}) => (
-  <FileInput
-    onChange={async file => {
-      const dirPath = creditApplicationTemplate.pathToSave
-      const filesCollection = client.collection('io.cozy.files')
+class EmptyDocumentHolder extends Component {
+  state = {
+    menuDisplayed: false
+  }
 
-      const classification = get(
-        creditApplicationTemplate.documents[documentId],
-        `rules.metadata.classification`
-      )
-      try {
-        let metadata = {}
-        if (classification) {
-          metadata = {
-            classification,
-            datetime: file.lastModifiedDate.toISOString()
-          }
-        }
-        const dirId = await filesCollection.ensureDirectoryExists(dirPath)
+  showMenu() {
+    this.setState({ menuDisplayed: true })
+  }
+  hideMenu() {
+    this.setState({ menuDisplayed: false })
+  }
 
-        const createdFile = await client
-          .collection('io.cozy.files')
-          .createFile(file, { dirId, metadata })
+  async onChange(file) {
+    const { documentId, client, linkDocumentSuccess } = this.props
+    const dirPath = creditApplicationTemplate.pathToSave
+    const filesCollection = client.collection('io.cozy.files')
 
-        linkDocumentSuccess({ document: createdFile.data, documentId })
-      } catch (uploadError) {
-        console.log('errr', uploadError)
+    const classification = get(
+      creditApplicationTemplate.documents[documentId],
+      `rules.metadata.classification`
+    )
+    try {
+      let metadata = {}
+      if (classification) {
+        metadata = {
+          classification,
+          datetime: file.lastModifiedDate.toISOString()
+        }
       }
-    }}
-    hidden={true}
-  >
-    <ButtonLink theme="ghost" extension="full" align="left">
-      <Icon icon="plus" size={16} className="u-mr-1 u-pa-half" />
-      <span>{t('documents.import')}</span>
-    </ButtonLink>
-  </FileInput>
-)
+      const dirId = await filesCollection.ensureDirectoryExists(dirPath)
+
+      const createdFile = await client
+        .collection('io.cozy.files')
+        .createFile(file, { dirId, metadata })
+
+      linkDocumentSuccess({ document: createdFile.data, documentId })
+    } catch (uploadError) {
+      console.log('errr', uploadError)
+    }
+  }
+
+  render() {
+    const { t, breakpoints } = this.props
+
+    const { isMobile } = breakpoints
+
+    if (isMobile) {
+      return (
+        <>
+          <Button
+            theme="ghost"
+            extension="full"
+            align="left"
+            onClick={() => this.showMenu()}
+          >
+            <Icon icon="plus" size={16} className="u-mr-1 u-pa-half" />
+            <span>{t('documents.import')}</span>
+          </Button>
+          {this.state.menuDisplayed && (
+            <ActionMenu onClose={() => this.hideMenu()}>
```

Comment:
I would prefer the action menu to be in a separate component

Reply:
Thanks! 

In the next PR we'll a WebMenu and a MobileActionMenu ;)